server/historian/packages/historian-base/src/services/restGitService.ts

```diff
@@ -265,37 +272,51 @@ export class RestGitService {
 		// In the future, we might want to also retrieve "channels". When that happens,
 		// our APIs will change so we specify what type we want to retrieve during
 		// the request.
-		const summaryCacheKey = this.getSummaryCacheKey("container");
-
-		// To do: Right now, we enable the cache anytime. We will choose either one of the following options:
-		// 1. Always using the cache but clearing the cache at session end from the server.
-		// 2. The driver fix to disable cache only on session start or cluster change if we can detect it.
-		// if (!useCache) {
-		//     // We set the useCache flag as false when we fetch the initial latest summary for a document.
-		//     // In that scenario, useCache as false means that we need to get the summary from storage and bypass the
-		//     // cache, since the cached data might be obsolete due to a cluster change. After fetching the value from
-		//     // storage, we add it to the cache so it can be up-to-date. As a result, subsequent requests that use
-		//     // the cache can read the correct value.
-		//     return this.fetchAndCache(summaryCacheKey, summaryFetch);
-		// }
+		const latestSummaryCacheKey = this.getSummaryCacheKey("container");
 
-		// If we get to this point, we want to obtain the latest summary, and we want to try reading it from the cache.
-		return this.resolve(summaryCacheKey, summaryFetch, true);
+		if (sha === LatestSummaryId) {
+			// Attempt to retrieve the latest summary from cache or storage, using specific `LatestSummaryCacheKey`.
+			// If retrieved from storage, this operation will also cache the retrieved value.
+			return this.resolve(latestSummaryCacheKey, summaryFetch, true);
+		}
+		// It is possible that the client will request the latest summary via specific sha instead of special "latest" sha.
+		// If we already have latest cached, we want to avoid hitting storage, so first check if the sha is the latest summary's sha.
+		// Important: we first check _only_ the sha in order to avoid unnecessarily reading an entire summary into memory from the cache.
+		const latestSummaryShaCacheKey = this.getSummaryCacheKey("container", LatestSummaryShaKey);
+		const cachedLatestSummarySha = await this.getCache<string | undefined>(
+			latestSummaryShaCacheKey,
+		);
+		if (cachedLatestSummarySha === sha) {
+			// If the requested sha is the same as the cached latest summary's sha, we should retrieve it
+			// from cache.
+			const cachedLatestSummary = await this.getCache<IWholeFlatSummary>(
+				latestSummaryCacheKey,
+			);
+			// If latest summary sha is cached, but the summary itself does not exist in cache, retrieve the requested summary
+			// by specific version as normal and do not cache it.
+			if (cachedLatestSummary) {
+				return cachedLatestSummary;
+			}
+		}
+		// We only cache the latest summary for each document, so if the requested summary is not latest,
+		// retrieve it without caching.
+		return summaryFetch();
```

Comment:
Could we instead use `resolve()` here?

Reply:
This is very intentionally _not_ resolve because we don't want to use the cache for this operation, and we don't want to store the result in cache either.